apt_info.py

```diff
@@ -69,7 +73,7 @@ def _write_pending_upgrades(registry, cache):
 def _write_held_upgrades(registry, cache):
     held_candidates = {
         p.candidate for p in cache
-        if p.is_upgradable and p._pkg.selected_state == apt_pkg.SELSTATE_HOLD
+        if p.is_upgradable and p._pkg.selected_state == apt_pkg.SELSTATE_HOLD and not p.phasing_applied
```

Comment:
linter also yells here, not sure how to fix that, perhaps:

```suggestion
        if (
            p.is_upgradable 
            and p._pkg.selected_state == apt_pkg.SELSTATE_HOLD 
            and not p.phasing_applied
        )
```